### DISCUSSION TITLE
Need some clarification about closing files when reading them for audio streams

### DISCUSSION THREAD

**Discussion by eliasdaler** [hajimehoshi/ebiten#1895] [Q&A] [answered]
Hello. I was looking at [this](https://github.com/hajimehoshi/ebiten/blob/main/examples/wav/main.go) example and noticed that the comment there says:

```go
        // In this example, embedded resource "Jab_wav" is used.
	//
	// If you want to use a wav file, open this and pass the file stream to wav.Decode.
	// Note that file's Close() should not be closed here
	// since audio.Player manages stream state.
```

Does this mean that it's not meant to be closed inside the `NewGame` function? Because the comments for `NewPlayer` say:

```go
// A Player doesn't close src even if src implements io.Closer.
// Closing the source is src owner's responsibility.
```

So, am I right to assume that if I use `os.File` for my wav sound, I still need to close it manually after the player is closed?

**Comment by hajimehoshi** [comment, marked as answer]
> So, am I right to assume that if I use os.File for my wav sound, I still need to close it manually after the player is closed?

Right. Ebiten doesn't care the audio source's closing state, so it's your responsibility to close the source.

**Reply by eliasdaler** [reply]
Thanks for the answer :)
